src/manifolds/MultinomialSymmetric.jl

```diff
@@ -0,0 +1,252 @@
+"""
+struct MultinomialSymmetric{N} <: AbstractEmbeddedManifold{ℝ,DefaultEmbeddingType} where {N} end
+
+function MultinomialSymmetric(n::Int)
+    return MultinomialSymmetric{n}()
+end
+
+@doc raw"""
+    check_manifold_point(M::MultinomialSymmetric, p)
+
+Checks whether `p` is a valid point on the [`MultinomialSymmetric`](@ref)`(m,n)` `M`,
+i.e. is a symmetric matrix with positive entries whose rows sum to one.
+"""
+check_manifold_point(::MultinomialSymmetric, ::Any)
+function check_manifold_point(M::MultinomialSymmetric{n}, p; kwargs...) where {n}
+    mpv =
+        invoke(check_manifold_point, Tuple{supertype(typeof(M)),typeof(p)}, M, p; kwargs...)
+    mpv === nothing || return mpv
+    r = sum(p, dims = 2)
+    if !isapprox(norm(r - ones(n, 1)), 0.0; kwargs...)
+        return DomainError(
+            r,
+            "The point $(p) does not lie on $M, since its rows do not sum up to one.",
+        )
+    end
+    if !(minimum(p) > 0)
+        return DomainError(
+            minimum(p),
+            "The point $(p) does not lie on $M, since at least one of its entries is nonpositive.",
+        )
+    end
+    return nothing
+end
+@doc raw"""
+    check_tangent_vector(M::MultinomialSymmetric p, X; check_base_point = true, kwargs...)
+
+Checks whether `X` is a valid tangent vector to `p` on the [`MultinomialSymmetric`](@ref) `M`.
+This means, that `p` is valid, that `X` is of correct dimension, symmetric, and sums to zero
+along any row.
+
+The optional parameter `check_base_point` indicates, whether to call
+[`check_manifold_point`](@ref check_manifold_point(::MultinomialSymmetric, ::Any))  for `p`.
+"""
+function check_tangent_vector(
+    M::MultinomialSymmetric{n},
+    p,
+    X;
+    check_base_point = true,
+    kwargs...,
+) where {n}
+    if check_base_point
+        mpe = check_manifold_point(M, p; kwargs...)
+        mpe === nothing || return mpe
+    end
+    mpv = invoke(
+        check_tangent_vector,
+        Tuple{supertype(typeof(M)),typeof(p),typeof(X)},
+        M,
+        p,
+        X;
+        check_base_point = false, # already checked above
+        kwargs...,
+    )
+    mpv === nothing || return mpv
+    r = sum(X, dims = 2) # due to symmetry, we only have to check columns
+    if !isapprox(norm(r), 0.0; kwargs...)
+        return DomainError(
+            r,
+            "The matrix $(X) is not a tangent vector to $(p) on $(M), since its columns/rows do not sum up to zero.",
+        )
+    end
+    return nothing
+end
+
+function decorated_manifold(::MultinomialSymmetric{N}) where {N}
+    return SymmetricMatrices(N, ℝ)
+end
+
+embed!(::MultinomialSymmetric, q, p) = copyto!(q, p)
+embed!(::MultinomialSymmetric, Y, ::Any, X) = copyto!(Y, X)
+
+@doc raw"""
+    inner(M::MultinomialSymmetric{n}, p, X, Y) where {n}
+
+Compute the inner product on the tangent space at $p$, which is the elementwise
+inner product similar to the [`Hyperbolic`](@ref) space, i.e.
```

Comment:
To me it looks more like the metric of `ProbabilitySimplex` or `MultinomialMatrices`.

Reply:
It's also `Hyperbolic(1)^(n x n)` :)

Reply:
For n=1 this says `<X, Y>_p = X[1]*X[1]/p[1]` while `Hyperbolic(1)` has `<X, Y>_p = -X[2]Y[2] + X[1]Y[1]`. I don't see much similarity here. Am I missing something?

Reply:
Oh! I mixed that up. It's `SymmetricPositiveDefinite(1,1)` (not `Hyperbolic`), i.e. element wise positive numbers. But also here we could embed into `MultinomialMatrices` to avoid implementing it again here.

Reply:
This is resolved by switching the embedding type.